src/features/search/redux/middleware.ts

```diff
@@ -17,6 +17,11 @@ export const searchMiddleware = ({ socketService }: MiddlewareDependencies): Mid
                 socket.emit('chart', [action.payload, range])
             }
 
+            if (isActionOf(updateQuery, action)) {
+                if (action.payload === '') dispatch(updateStockList([]))
+                socket.emit('search', action.payload)
```

Comment:
why `action.payload === '' `?

Reply:
so when they delete whatever they have entered in query (input) stocklist state is changed to [] as we know nothing = nothing